code/modules/mob/living/simple_animal/slime/life.dm

```diff
@@ -129,8 +134,27 @@
 	else
 		Tempstun = 0
 
+	if(stat != DEAD)
+		var/co2_percentage =0
+		if(("co2" in environment.gases) && environment.total_moles())
```

Comment:
You don't need to check for total_moles() here; if "co2" exists then there is a guarantee that there will be a non-zero amount of gas in the mixture.